plugins/woocommerce/tests/php/src/Internal/Admin/ReviewsCommentsOverridesTest.php

```diff
@@ -12,6 +13,64 @@
  */
 class ReviewsCommentsOverridesTest extends WC_Unit_Test_Case {
 
+	/**
+	 * @covers \Automattic\WooCommerce\Internal\Admin\ReviewsCommentsOverrides::display_notices()
+	 * @dataProvider provider_test_display_notices()
+	 * @param string $current_screen_base    The current WP_Screen base value.
+	 * @param bool   $should_display_notices Whether notices should be displayed.
+	 */
+	public function test_display_notices( string $current_screen_base, bool $should_display_notices ) {
+		global $current_screen;
+
+		$screen = new \stdClass();
+		$screen->base = $current_screen_base;
+
+		$current_screen = $screen; // phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+
+		ob_start();
+
+		ReviewsCommentsOverrides::get_instance()->display_notices();
+
+		$output = ob_get_clean();
+
+		if ( $should_display_notices ) {
+			$this->assertNotEmpty( $output );
+		} else {
+			$this->assertEmpty( $output );
+		}
+	}
+
+	/** @see test_display_notices() */
+	public function provider_test_display_notices() : \Generator {
+		yield 'Comments page' => [ 'edit-comments', true ];
+		yield 'Posts page' => [ 'edit', false ];
+		yield 'Product Reviews page' => [ 'product_page_product-reviews', false ];
```

Comment:
What if we test here with a small HTML code snippet rather than just checking if the `$output` is empty or not?

Reply:
It would make this test depend on the other method. I know it already does, but right now it just depends on the other method outputting _something_, it doesn't care what. Ideally, we would be mocking the other method and testing this one in isolation. Well, that if it was really a unit test. Our tests for this epic are in a very weird place between unit and integration tests...